sdlib/d/gc/page.d

```diff
@@ -726,9 +745,24 @@ private:
 					auto newOccupancy = oldOccupancy & bmp[i];
 
 					occupancyMask |= newOccupancy;
-					count += popCount(oldOccupancy ^ newOccupancy);
+					auto toRemove = oldOccupancy ^ newOccupancy;
+					count += popCount(toRemove);
+
+					scope(success) e.slabData.rawContent[i] = newOccupancy;
+
+					if (!ec.supportsMetadata) {
+						continue;
+					}
+
+					toRemove &= e.slabMetadataFlags.rawContent[i];
 
-					e.slabData.rawContent[i] = newOccupancy;
+					if (toRemove) {
+						import d.gc.slab;
+						auto ssize = binInfos[sc].slotSize;
```

Comment:
Yes, I could do this. I first avoided it, because the second case does not have a nimble offset. But I can just push that mechanism into the function.